src/styles/utility/_spacing.scss

```diff
@@ -0,0 +1,126 @@
+/*
+Generates utility classes for spacing (margin and padding)
+
+The convention is ".bdl-py-0"
+  .bdl- prefix
+  m margin, p padding
+  null, x, y, t, r, b, l (sides)
+  0,1,2... multiplier for grid units
+
+Margins
+  .bdl-m-0, .bdl-mx-1, .bdl-my-2, .bdl-mt-3, .bdl-mr-4, etc.
+
+Padding
+  .bdl-p-0, .bdl-px-1, .bdl-py-2, .bdl-pt-3, .bdl-pr-4, etc.
+*/
+@import '../constants/layout';
+@import './generate-utility';
+
+$spacer: $grid-unit * 4 !default;
+$spacers: () !default;
+$spacers: map-merge(
+    (
+        0: 0,
+        1: $spacer * .25,
+        2: $spacer * .5,
+        3: $spacer,
+        4: $spacer * 1.5,
+        5: $spacer * 3,
+    ),
+    $spacers
+);
+
+$spacing-utilities: () !default;
+$spacing-utilities: map-merge(
+    (
+        "margin": (
+            responsive: false,
+            property: margin,
+            class: m,
+            values: map-merge($spacers, (auto: auto))
+        ),
+        "margin-x": (
+            responsive: false,
+            property: margin-right margin-left,
+            class: mx,
```

Comment:
I think maybe we can get rid of X and Y. And force people to be explicit. Unless there are use cases were we do this often.

Reply:
x & y (v and h) are very useful actually, every utility framework supports them

For example: https://tailwindcss.com/docs/padding/